src/core/src/texture.rs

```diff
@@ -314,6 +314,18 @@ impl Kind {
     }
 }
 
+/// The marker for the texture initializer to generate extra space for the mipmap generation.
+#[allow(missing_docs)]
+#[derive(Clone, Copy, Debug, Eq, Hash, Ord, PartialEq, PartialOrd)]
+#[cfg_attr(feature = "serialize", derive(Serialize, Deserialize))]
+#[repr(u8)]
+pub enum Mipmap {
```

Comment:
I don't think you need to pass `Option<Mipmap>` around as opposed to just `Mipmap`, seems to unnecessarily complicate things